workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerWorkerConnector.cs

```diff
@@ -1,18 +1,86 @@
 using System;
+using System.Threading.Tasks;
 using Fps;
 using Improbable.Gdk.Core;
 using Improbable.Gdk.GameObjectCreation;
 using Improbable.Gdk.GameObjectRepresentation;
 using Improbable.Gdk.PlayerLifecycle;
+using Improbable.Worker.CInterop;
+using Improbable.Worker.CInterop.Alpha;
+using UnityEngine;
 
 public class SimulatedPlayerWorkerConnector : DefaultWorkerConnector
 {
     private const string AuthPlayer = "Prefabs/SimulatedPlayer/SimulatedPlayer";
     private const string NonAuthPlayer = "Prefabs/SimulatedPlayer/SimulatedPlayerProxy";
 
-    private async void Start()
+    private ILogDispatcher simulatedCoordinatorLogDispatcher;
+    private bool connectToRemoteDeployment;
+    private string loginToken;
+    private string playerIdentityToken;
+
+    public async Task ConnectSimulatedPlayer(ILogDispatcher logDispatcher, string SimulatedPlayerDevAuthTokenId, string SimulatedPlayerTargetDeployment)
     {
-        await Connect(WorkerUtils.SimulatedPlayer, new ForwardingDispatcher());
+        simulatedCoordinatorLogDispatcher = logDispatcher;
+
+        // If we're connecting simulated players to another deployment, we need to fetch locator tokens.
+        if (!string.IsNullOrEmpty(SimulatedPlayerTargetDeployment))
+        {
+            if (string.IsNullOrEmpty(SimulatedPlayerDevAuthTokenId))
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent("Failed to launch simulated player. No development authentication token specified."));
+                return;
+            }
+
+            // Obtain player identity token for simulated player.
+            var playerIdentityTokenResponse = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync("locator.improbable.io", 444,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationTokenId = SimulatedPlayerDevAuthTokenId,
+                    DisplayName = "",
+                    PlayerId = "SimulatedPlayer"
+                }).Get();
+            if (playerIdentityTokenResponse.Value.Status != ConnectionStatusCode.Success)
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent("Failed to launch simulated player. Failed to obtain player identity token. Reason: " + playerIdentityTokenResponse?.Error));
```

Comment:
Use string interpolation. Maybe move the reason to its own log field?

Reply:
I'd argue that string interpolation shouldn't really be used inside top level messages. They should always be literals. The dynamic data should be fields inside it.

Reply:
This is now handled by the base class impl